data-prepper-plugins/dynamodb-source-coordination-store/src/test/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbClientWrapperTest.java

```diff
@@ -330,52 +347,175 @@ void getSourcePartitionItem_catches_exception_from_getItem_and_returns_empty_opt
 
         assertThat(result.isPresent(), equalTo(true));
-        assertThat(result.get(), equalTo(dynamoDbSourcePartitionItemPageIterable));
+        assertThat(result.get(), equalTo(acquiredItem));
+
+        verify(acquiredItem).setPartitionOwner(ownerId);
+        verify(acquiredItem).setSourcePartitionStatus(SourcePartitionStatus.ASSIGNED);
+        verify(acquiredItem).setSourceStatusCombinationKey(sourceIdentifier + "|" + SourcePartitionStatus.ASSIGNED);
+
+        final ArgumentCaptor<Instant> partitionOwnershipArgumentCaptor = ArgumentCaptor.forClass(Instant.class);
+
+        verify(acquiredItem).setPartitionOwnershipTimeout(partitionOwnershipArgumentCaptor.capture());
 
-        final ScanEnhancedRequest scanEnhancedRequest = requestArgumentCaptor.getValue();
+        final Instant newPartitionOwnershipTimeout = partitionOwnershipArgumentCaptor.getValue();
 
-        assertThat(scanEnhancedRequest.filterExpression(), equalTo(filterExpression));
-        assertThat(scanEnhancedRequest.limit(), equalTo(20));
+        assertThat(newPartitionOwnershipTimeout.isAfter(now.plus(ownershipTimeout)), equalTo(true));
+
+        verify(acquiredItem).setPartitionPriority(newPartitionOwnershipTimeout.toString());
     }
 
-    @ParameterizedTest
-    @MethodSource("exceptionProvider")
-    void getSourcePartitionItems_catches_exception_from_scan_and_returns_empty_optional(final Class exception) throws NoSuchFieldException, IllegalAccessException {
+    @Test
+    void getAvailablePartition_with_assigned_partition_with_unexpired_partitionOwnershipTimeout_returns_empty_optional() throws NoSuchFieldException, IllegalAccessException {
+        final String ownerId = UUID.randomUUID().toString();
+        final Duration ownershipTimeout = Duration.ofMinutes(1);
+        final String sourceStatusCombinationKey = sourceIdentifier + "|" + SourcePartitionStatus.ASSIGNED;
+
         final DynamoDbTable<DynamoDbSourcePartitionItem> table = mock(DynamoDbTable.class);
-        doThrow(exception).when(table).scan(any(ScanEnhancedRequest.class));
+        final DynamoDbIndex<DynamoDbSourcePartitionItem> sourceStatusIndex = mock(DynamoDbIndex.class);
+        given(table.index(SOURCE_STATUS_COMBINATION_KEY_GLOBAL_SECONDARY_INDEX)).willReturn(sourceStatusIndex);
+
+        final DynamoDbSourcePartitionItem firstItem = mock(DynamoDbSourcePartitionItem.class);
+        final DynamoDbSourcePartitionItem secondItem = mock(DynamoDbSourcePartitionItem.class);
+        given(firstItem.getPartitionOwnershipTimeout()).willReturn(Instant.now().plus(2, ChronoUnit.MINUTES));
 
-        final Expression filterExpression = mock(Expression.class);
+        final SdkIterable<Page<DynamoDbSourcePartitionItem>> pageSdkIterable = () -> {
+            final Page<DynamoDbSourcePartitionItem> emptyPage = mock(Page.class);
+            given(emptyPage.items()).willReturn(List.of(firstItem, secondItem));
```

Comment:
This page is not empty. It contains 2 items.

Reply:
Copy paste error